bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/ComponentUtils.java

```diff
@@ -166,4 +177,103 @@ public static ContentPolicy getPolicy(String contentPath, ResourceResolver resou
         return policy;
     }
 
+    /**
+     * Takes in a single variable and returns an array of size 1 consisting of the same variable.
+     * Useful to directly iterate on, when something returns either a single value or an array of values of a particular type.
+     *
+     * @param param of any type T
+     * @return array of size 1 having the same variable
+     */
+    public static <T> T[] convertToArray(T param) {
+        if (param.getClass().isArray()) {
+            return (T[]) param;
+        } else {
+            T[] array = (T[]) Array.newInstance(param.getClass(), 1);
+            array[0] = (T) param;
+            return array;
+        }
+    }
+
+    /**
+     * Fetches all the custom property group names defined in the template policy
+     *
+     * @param policy reference to the template policy {@link ContentPolicy}
+     * @param resourceResolver {@link ResourceResolver}
+     * @return {@code List<String>} a list of all the group names of custom properties defined in the policy
+     */
+    public static List<String> getCustomPropertyGroupsFromPolicy(ContentPolicy policy, ResourceResolver resourceResolver) {
+        if (policy == null) {
+            return new ArrayList<>();
+        }
+        Resource policyResource = resourceResolver.resolve(policy.getPath());
+        List<String> groupNames = new ArrayList<>();
+        List<Resource> customPropertiesGroupsList;
```

Comment:
you cam declare this variable inside if block 
`if (customPropertiesResource != null) {`